taf/developer_tool.py

```diff
@@ -134,27 +154,107 @@ def build_auth_repo(
+            while existing_key is None:
+                for existing_role_name, role_keys in yubikeys.items():
+                    if existing_role_name == role_name:
+                        continue
+                    print(f"Existing keys for role {existing_role_name} are:\n")
+                    for key_and_cert in role_keys.values():
+                        key, cert_cn = key_and_cert
+                        key_id_certs[key["keyid"]] = cert_cn
+                        print(f"{cert_cn} id: {key['keyid']}")
+                existing_keyid = input("\nEnter existing Yubikey's id and press ENTER ")
+                try:
+                    existing_key = get_key(existing_keyid)
+                    cert_cn = key_id_certs[existing_keyid]
+                except UnknownKeyError:
+                    pass
+    if not use_existing:
+        input("Please insert a new YubiKey and press ENTER.")
+        serial_num = yk.get_serial_num()
+        while serial_num in yubikeys[role_name]:
+            print(f"Yubikey with serial number {serial_num} is already in use.\n")
+            input("Please insert new YubiKey and press ENTER.")
+            serial_num = yk.get_serial_num()
+
+        pin = get_pin_for(key_name)
+
+        cert_cn = input("Enter key holder's name: ")
+
+        print("Generating keys, please wait...")
+        pub_key_pem = yk.setup(pin, cert_cn, cert_exp_days=EXPIRATION_INTERVAL).decode(
+            "utf-8"
+        )
+
+        key = import_rsakey_from_pem(pub_key_pem, scheme)
+
+        cert_path = Path(certs_dir, f"{key['keyid']}.cert")
+        with open(cert_path, "wb") as f:
+            f.write(yk.export_piv_x509())
+
+    # set Yubikey expiration date
+    role_obj.add_verification_key(key, expires=YUBIKEY_EXPIRATION_DATE)
+    role_obj.add_external_signature_provider(
+        key, partial(signature_provider, key["keyid"], cert_cn)
+    )
+    yubikeys[role_name][serial_num] = (key, cert_cn)
+
+
+def _register_key(keystore, passwords, role_obj, role_name, key_name, key_num, scheme):
+    # if keystore exists, load the keys
+    # this is useful when generating tests
+    if keystore is not None:
+        public_key = import_rsa_publickey_from_file(
+            os.path.join(keystore, f"{key_name}.pub"), scheme
+        )
+        password = passwords[key_num]
+        if password:
+            private_key = import_rsa_privatekey_from_file(
+                os.path.join(keystore, key_name), password, scheme=scheme
+            )
+        else:
+            private_key = import_rsa_privatekey_from_file(
+                os.path.join(keystore, key_name), scheme=scheme
+            )
+
+    # if it does not, generate the keys and print the output
+    else:
+        key = generate_rsa_key()
+        print(f"{role_name} key:\n\n{key['keyval']['private']}\n\n")
+        public_key = private_key = key
+    role_obj.add_verification_key(public_key)
+    role_obj.load_signing_key(private_key)
+
+
 def create_repository(
```

Comment:
Definitely nothing crucial, but we there is one `os.path.isdir` left in this function